src/shared/directives/conditional-show/services/condition-parser.service.ts

```diff
@@ -0,0 +1,204 @@
+      return false;
+    }
+    const formatExpectedValue = expectedValue ? expectedValue.toString().trim() : '';
+    return formatCurrentValue != formatExpectedValue; // tslint:disable-line
+  }
+
+  private static checkMultiSelectListEquals(expectedValue: string, currentValue: any, conditionSeparator: string): boolean {
+    const expectedValues = expectedValue.split(',').sort().toString();
+    const values = currentValue ? currentValue.sort().toString() : '';
+    if (conditionSeparator === ShowCondition.CONDITION_NOT_EQUALS) {
+      return expectedValues !== values;
+    } else {
+      return expectedValues === values;
+    }
+  }
+
+  private static checkValueContains(expectedValue: string, currentValue: any): boolean {
+    if (expectedValue.search(',') > -1) {
+      let expectedValues = expectedValue.split(',').sort();
+      let values = currentValue ? currentValue.sort().toString() : '';
+      return expectedValues.every(item => values.search(item) >= 0);
+    } else {
+      let values = currentValue && Array.isArray(currentValue) ? currentValue.toString() : '';
+      return values.search(expectedValue) >= 0;
+    }
+  }
+
+  private static unquoted(str: string): string {
+    const res = str.replace(/^"|"$/g, '');
+    return res;
+  }
+
+  private static findValueForComplexCondition(fields: object, head: string, tail: string[], path?: string): any {
+    if (!fields) {
+      return undefined;
+    }
+    if (tail.length === 0) {
+      return this.getValue(fields, head);
+    } else {
+      if (FieldsUtils.isArray(fields[head])) {
+        return this.findValueForComplexConditionInArray(fields, head, tail, path);
+      } else {
+        return this.findValueForComplexConditionForPathIfAny(fields, head, tail, path);
+      }
+    }
+  }
+
+  private static findValueForComplexConditionForPathIfAny(fields: object, head: string, tail: string[], path?: string): any {
+    if (path) {
+      const [_, ...pathTail] = path.split(/[_]+/g);
+      return this.findValueForComplexCondition(fields[head], tail[0], tail.slice(1), pathTail.join('_'));
+    } else {
+      return this.findValueForComplexCondition(fields[head], tail[0], tail.slice(1), path);
+    }
+  }
+
+  private static findValueForComplexConditionInArray(fields: object, head: string, tail: string[], path?: string): any {
+  // use the path to resolve which array element we refer to
+    if (path.startsWith(head)) {
+      const [_, ...pathTail] = path.split(/[_]+/g);
+      if (pathTail.length > 0) {
+        try {
+          let arrayIndex = Number.parseInt(pathTail[0], 10);
+          const [__, ...dropNumberPath] = pathTail;
+          return (fields[head][arrayIndex] !== undefined) ? this.findValueForComplexCondition(
+          fields[head][arrayIndex]['value'], tail[0], tail.slice(1), dropNumberPath.join('_')) : null;
+        } catch (e) {
+          console.log('Error while parsing number', pathTail[0], e);
```

Comment:
should this be console.error

Reply:
This is existing code I just moved to this file so want to keep it as is

Reply:
if this is copied from exsiting code, please also change it.

Reply:
changed to console.error